### PR TITLE
Remove mechanism to hide Exception tree

### PR DESCRIPTION
**Pull request by sebr72**

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java [hunk 1]
```diff
@@ -286,8 +286,7 @@ private String validateURL(final String externalGraphicUrl) {
         final URL fileURL = new URL("file://" + externalGraphicUrl);
         return testForLegalFileUrl(this.configuration, fileURL).toExternalForm();
       } catch (MalformedURLException e1) {
-        // unable to convert to file url so give up and throw exception;
-        throw ExceptionUtils.getRuntimeException(e);
+        throw new PrintException("Unable to convert to file url " + externalGraphicUrl, e);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java:289]:
```suggestion
        throw new PrintException("Unable to convert to file URL " + externalGraphicUrl, e);
```

File: core/src/main/java/org/mapfish/print/servlet/MapPrinterServlet.java [hunk 2]
```diff
@@ -295,7 +295,7 @@ private static PJsonObject parseJson(
         try {
           requestData = URLDecoder.decode(requestData, Constants.DEFAULT_ENCODING);
         } catch (UnsupportedEncodingException e) {
-          throw ExceptionUtils.getRuntimeException(e);
+          throw new PrintException("Failed to decode using " + Constants.DEFAULT_ENCODING, e);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/MapPrinterServlet.java:298]: No need to add requestData?

File: core/src/main/java/org/mapfish/print/servlet/MapPrinterServlet.java [hunk 3]
```diff
@@ -308,7 +308,7 @@ private static PJsonObject parseJson(
         try {
           return MapPrinter.parseSpec(URLDecoder.decode(requestData, Constants.DEFAULT_ENCODING));
         } catch (UnsupportedEncodingException uee) {
-          throw ExceptionUtils.getRuntimeException(e);
+          throw new PrintException("Failed to decode using " + Constants.DEFAULT_ENCODING, e);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/servlet/MapPrinterServlet.java:311]: No need to add requestData?

File: core/src/main/java/org/mapfish/print/URIUtils.java [hunk 4]
```diff
@@ -157,7 +157,7 @@ private static boolean addParams(
         result.append("=");
         result.append(URLEncoder.encode(val, Constants.DEFAULT_ENCODING));
       } catch (UnsupportedEncodingException e) {
-        throw ExceptionUtils.getRuntimeException(e);
+        throw new PrintException("Failed to encode using " + Constants.DEFAULT_ENCODING, e);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/URIUtils.java:160]: No need to add val?

**Review by sbrunner — APPROVED** [review]
Just some small comments, thanks for this good cleanup :-)

File: core/src/main/java/org/mapfish/print/URIUtils.java (hunk 4 above)
Comment (sebr72) [reply core/src/main/java/org/mapfish/print/URIUtils.java:160]: it is not always val failing. So I changed the code to report the correct issue.

**Pull request merged**